### PR TITLE
Added new kafka backend

### PR DESCRIPTION
**Pull request by StasEvseev**
Made metric agent by configurable as usual django configs.

### CODE REVIEW

File: requirements/requirements-testing.txt [hunk 1]
```diff
@@ -7,7 +7,8 @@ freezegun>=0.3.7,<0.4
 mock>=2.0,<3.0
 
 # PyTest for running the tests.
-pytest
+#: pinned until pytest-asyncio will be compatible with pytest==`4.1.0`
```
Comment (kammala) [inline requirements/requirements-testing.txt]: looks like it [already is](https://github.com/pytest-dev/pytest-asyncio#0100-2019-01-08)

File: time_execution/backends/kafka.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        if self._producer:
+            return self._producer
+
+        try:
+            self._producer = KafkaProducer(
+                bootstrap_servers=self.hosts,
+                value_serializer=lambda v: JSONSerializer().dumps(v).encode('utf-8'),
+                **self._kwargs
+            )
+        except NoBrokersAvailable:
```
[the remark this answers is not in the document]
Comment (StasEvseev) [reply time_execution/backends/kafka.py]: Good one! Probably leftovers from rewriting client(I did it couple of times).

File: time_execution/backends/kafka.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            data (dict): Additional data to store with the metric
+        """
+
+        data["name"] = name
+        if not ("timestamp" in data):
+            data["timestamp"] = datetime.utcnow()
+
+        try:
+            self.producer.send(topic=self.topic, value=data)
+            self.producer.flush()
```
[the remark this answers is not in the document]
Comment (StasEvseev) [reply time_execution/backends/kafka.py]: Yes. That is correct. The thread will be blocked. I assume that client will be run in the `ThreadedBackend`

File: requirements/requirements-testing.txt (hunk 1 above)
Comment (StasEvseev) [reply requirements/requirements-testing.txt]: Will clean it up!

File: README.rst [hunk 4]
```diff
@@ -30,6 +30,7 @@ Backends
 
 - InfluxDB 0.8
 - Elasticsearch 5
+- Kafka
```
[the remark this answers is not in the document]
Comment (StasEvseev) [reply README.rst]: I did. The only thing that I don't like in the one you mentioned that it requires a build dependencies(gcc, librdkafka). I just picked up fully python based one.

File: time_execution/serializer.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+"""
+Module copied from elasticsearch package to serialize data in same way as elasticsearch does.
```
Comment (igbun) [inline time_execution/serializer.py]: Perhaps good to include a link to the repo of elasticsearch-py.

**Pull request merged**